service/controller/v2/resources/vaultcrt/current.go

```diff
@@ -67,6 +75,17 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	return secret, nil
 }
 
+// checkCertType checks whether customObject is one of the Cert types we are supporting in certs library.
+func (r *Resource) checkCertType(customObject v1alpha1.CertConfig) bool {
+	c := certs.Cert(customObject.Spec.Cert.ClusterComponent)
```

Comment:
```suggestion
	c := certs.Cert(key.ClusterComponent(customObject))
```

I think we should have a key function for this and also use it in the logging.